indico/modules/users/controllers.py

```diff
@@ -738,6 +740,24 @@ def _process_pending_users(self, results):
             if ext_id is not None:
                 cache.set(ext_id, self.externals[ext_id], timeout=86400)
 
+    @use_kwargs({
+        'category_id': fields.Integer(validate=lambda x: x >= 0),
+        'event_id': fields.Integer(validate=lambda x: x >= 0)
```

Comment:
I'd skip the validate here; we don't use this for any other ID fields either (searched for `_id': fields.Int`).

*If* we wanted this level of validation I'd rather add a new custom type like `DatabaseID` which does the validation - but I think that's not really necessary since a negative ID is not invalid per se, it's just an ID that won't have a corresponding object.